### PR TITLE
Docs for MXRecordIO, MXIndexedRecordIO modified

### PR DESCRIPTION
**Pull request by Roshrini**
@mli @zackchase @madjam @nswamy @indhub @jiajiechen Please take a look.

### CODE REVIEW

File: python/mxnet/recordio.py [hunk 1]
```diff
@@ -132,31 +138,75 @@ def open(self):
...
+
+        This function is internally called by `read_idx` to find the current
+        reader pointer position. It doesn't return anything."""
         assert not self.writable
         pos = ctypes.c_size_t(self.idx[idx])
         check_call(_LIB.MXRecordIOReaderSeek(self.handle, pos))
 
     def tell(self):
-        """Query current write head position."""
+        """Queries current writer pointer position and returns it.
```
Comment (madjam) [inline python/mxnet/recordio.py]: Suggest rewording it as:
"Returns the current position of write head."

File: python/mxnet/recordio.py [hunk 2]
```diff
@@ -132,31 +138,75 @@ def open(self):
...
+        80
+        """
         assert self.writable
         pos = ctypes.c_size_t()
         check_call(_LIB.MXRecordIOWriterTell(self.handle, ctypes.byref(pos)))
         return pos.value
 
     def read_idx(self, idx):
-        """Read record with index."""
+        """Reads the record on the given index and returns it.
```
Comment (madjam) [inline python/mxnet/recordio.py]: Suggest rewording as:
"Returns record at given index."

File: python/mxnet/recordio.py [hunk 3]
```diff
@@ -132,31 +138,75 @@ def open(self):
...
+        >>> record = mx.recordio.MXIndexedRecordIO('tmp.idx', 'tmp.rec', 'r')
+        >>> record.read_idx(3)
+        record_3
+        """
         self.seek(idx)
         return self.read()
 
     def write_idx(self, idx, buf):
-        """Write record with index."""
+        """Writes the record on the given index.
```
Comment (madjam) [inline python/mxnet/recordio.py]: "Inserts record at given index."

**Comment by Roshrini** [comment]
Fixed

File: python/mxnet/recordio.py [hunk 4]
```diff
@@ -17,14 +17,19 @@
     cv2 = None
 
 class MXRecordIO(object):
-    """Read/write RecordIO format data.
+    """Reads/writes `RecordIO` data format, supporting sequential read and write.
+
+    Example usage:
+    ----------
+    >>> record = mx.recordio.MXRecordIO('tmp.rec', 'w')
```
Comment (piiswrong) [inline python/mxnet/recordio.py]: put a read/write example here.

File: python/mxnet/recordio.py [hunk 5]
```diff
@@ -132,31 +178,75 @@ def open(self):
...
     def close(self):
+        """Closes the record file."""
         if not self.is_open:
             return
         super(MXIndexedRecordIO, self).close()
         self.fidx.close()
 
     def seek(self, idx):
-        """Query current read head position."""
+        """Sets the current reader pointer position.
```
Comment (madjam) [inline python/mxnet/recordio.py]: "current reader" -> read

File: python/mxnet/recordio.py (hunk 4 above)
Comment (Roshrini) [inline python/mxnet/recordio.py]: read/write example is added in the respective function. If I add it here, it might be redundant? I added whole example now so that one should able to run it.

File: python/mxnet/recordio.py [hunk 6]
```diff
@@ -132,31 +198,75 @@ def open(self):
...
+        80
+        """
         assert self.writable
         pos = ctypes.c_size_t()
         check_call(_LIB.MXRecordIOWriterTell(self.handle, ctypes.byref(pos)))
         return pos.value
 
     def read_idx(self, idx):
-        """Read record with index."""
+        """Returns a record at given index.
```
Comment (zackchase) [inline python/mxnet/recordio.py]: Perhaps change "a" to "the" since there is only one record at the given index. Is that right?

**Comment by Roshrini** [comment]
@zackchase Thanks for reviewing. Fixed it. I added examples after running them. But @madjam can run them again. Please approve and merge.

File: python/mxnet/recordio.py [hunk 7]
```diff
@@ -98,16 +151,28 @@ def read(self):
             return None
 
 class MXIndexedRecordIO(MXRecordIO):
-    """Read/write RecordIO format data supporting random access.
+    """Reads/writes `RecordIO` data format, supporting random access.
+
+    Example usage:
+    ----------
+    >>> record = mx.recordio.MXIndexedRecordIO('tmp.idx', 'tmp.rec', 'w')
+    >>> record
```
Comment (piiswrong) [inline python/mxnet/recordio.py]: remove these two lines
